app/src/main/java/com/hedvig/app/feature/marketpicker/LanguageAdapter.kt

```diff
@@ -0,0 +1,123 @@
+                            language.text =
+                                language.resources.getString(R.string.SETTINGS_LANGUAGE_SWEDISH)
+                            parent.setHapticClickListener {
+                                languageViewModel.selectLanguage(Language.SV_SE)
+                            }
+                        }
+                    }
+                    Market.NO -> {
+                        holder.apply {
+                            language.text = language.resources.getString(R.string.norwegian)
+                            parent.setHapticClickListener {
+                                languageViewModel.selectLanguage(Language.NB_NO)
+
+                            }
+                        }
+                    }
+                }
+            }
+            EN -> holder.apply {
+                language.text =
+                    language.resources.getString(R.string.SETTINGS_LANGUAGE_ENGLISH)
+            }
+        }
+
+        if (position == 0 && holder.button.isChecked) {
+            lastChecked = holder.button
+            lastCheckedPos = 0
+        }
+
+        holder.parent.setHapticClickListener { v ->
+            when (position) {
+                EN -> languageViewModel.selectLanguage(Language.EN_SE)
+                LOCAL -> {
+                    when (selectedMarket) {
+                        Market.SE -> languageViewModel.selectLanguage(Language.SV_SE)
+                        Market.NO -> languageViewModel.selectLanguage(Language.NB_NO)
+                    }
+                }
+            }
+            val rb = v.radioButton as RadioButton
+            rb.isChecked = true
+            rb.background = rb.context.getDrawable(R.drawable.ic_radiob_button_checked)
+            animateRadioButton(holder)
+            if (rb.isChecked) {
+                if (lastChecked != null) {
+                    if (lastCheckedPos != position) {
+                        lastChecked!!.background =
+                            rb.context.getDrawable(R.drawable.ic_radio_button_unchecked)
+                        lastChecked!!.isChecked = false
+                        animateRadioButton(holder)
+                    }
+                }
+                lastChecked = rb
+                lastCheckedPos = position
+            } else lastChecked = null
+        }
+    }
+
+    private fun animateRadioButton(holder: ViewHolder) {
+        val button = holder.button
+        button.scaleX = 0f
+        button.scaleY = 0f
+        button.spring(
+            SpringAnimation.SCALE_X,
+            SpringForce.STIFFNESS_HIGH,
+            SpringForce.DAMPING_RATIO_NO_BOUNCY
+        ).animateToFinalPosition(1f)
+        button.spring(
+            SpringAnimation.SCALE_Y,
+            SpringForce.STIFFNESS_HIGH,
+            SpringForce.DAMPING_RATIO_NO_BOUNCY
+        ).animateToFinalPosition(1f)
+    }
+
+    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        val parent = view
```

Comment:
```suggestion
        val parent = view
```

We can use `itemView` here 🙂